salt/state.py

```diff
@@ -2813,29 +2855,15 @@ def render_state(self, sls, saltenv, mods, matches, local=False):
                         continue
 
                     if inc_sls.startswith('.'):
-                        match = re.match(r'^(\.+)(.*)$', inc_sls)
-                        if match:
-                            levels, include = match.groups()
-                        else:
-                            msg = ('Badly formatted include {0} found in include '
-                                    'in SLS \'{2}:{3}\''
-                                    .format(inc_sls, saltenv, sls))
-                            log.error(msg)
-                            errors.append(msg)
-                            continue
-                        level_count = len(levels)
-                        p_comps = sls.split('.')
-                        if state_data.get('source', '').endswith('/init.sls'):
-                            p_comps.append('init')
-                        if level_count > len(p_comps):
-                            msg = ('Attempted relative include of \'{0}\' '
-                                   'within SLS \'{1}:{2}\' '
-                                   'goes beyond top level package '
-                                   .format(inc_sls, saltenv, sls))
-                            log.error(msg)
-                            errors.append(msg)
+                        try:
+                            source_file = state_data.get('source', '')
+                            _sls = sls
+                            if source_file.endswith('/init.sls'):
+                                _sls += '.init'
```

Comment:
So you are working with the source file here right? thats what you need out of the state_data? In the `fn_` variable not what you need?